### DISCUSSION TITLE
Alllow store.$reset pass the keypath as the argument to reset nested value

### DISCUSSION THREAD

**Discussion by mockingjet** [vuejs/pinia#679] [Ideas]
```js
const useDataStore = defineStore('date', {
  state: ()=> {
    return {
      data: [],
      pagination: {
        page: 1,
        size: 15,
        total: 0,
      },
    };
  },
}
```

Allowing to use dataStore.$reset('pagination') or dataStore.$reset('pagintion.page') can increase developer experience

**Comment by posva** [comment]
This would be better as a plugin. Currently the reset functionality is very limited and that's intentional: it only work with the options api and simply calls the state function again.

**Comment by JiatLn** [comment]
Hi~ I wrote a plugin to support your idea. you could try it.
[pinia-plugin-keep](https://github.com/JiatLn/pinia-plugin-keep)
